src/emulator/gui/src/settings_dialog.cpp

```diff
@@ -48,13 +49,83 @@ bool change_pref_location(const std::string &input_path, const std::string &curr
+                "libnaac", "libnet", "libnetctl", "libngs", "libpaf", "libpaf_web_map_view", "libperf", "libpgf", "libpvf", "librudp",
+                "libsas", "libsceavplayer", "libSceBeisobmf", "libSceBemp2sys", "libSceCompanionUtil", "libSceDtcpIp", "libSceFt2",
+                "libscejpegarm", "libscejpegencarm", "libSceJson", "libscemp4", "libSceMp4Rec", "libSceMusicExport", "libSceNearDialogUtil",
+                "libSceNearUtil", "libScePhotoExport", "libScePromoterUtil", "libSceScreenShot", "libSceShutterSound", "libSceSqlite",
+                "libSceTelephonyUtil", "libSceTeleportClient", "libSceTeleportServer", "libSceVideoExport", "libSceVideoSearchEmpr", 
+                "libSceXml", "libshellsvc", "libssl", "libsystemgesture", "libult", "libvoice", "libvoiceqos", "livearea_util",
+                "mail_api_for_local_libc", "near_profile", "notification_util", "np_activity", "np_activity_sdk", "np_basic",
+                "np_commerce2", "np_common", "np_common_ps4", "np_friend_privacylevel", "np_kdc", "np_manager", "np_matching2",
+                "np_message", "np_message_contacts", "np_message_dialog_impl", "np_message_padding", "np_party", "np_ranking", 
+                "np_signaling",    "np_sns_facebook", "np_trophy", "np_tus", "np_utility", "np_webapi", "party_member_list",
+                "psmkdc", "pspnet_adhoc", "signin_ext", "sqlite", "store_checkout_plugin", "trigger_util", "web_ui_plugin",
+                // clang-format on
+            };
+            ImGui::PushItemWidth(-1);
+            if (ImGui::ListBoxHeader("##modules_list", IM_ARRAYSIZE(list_modules), 8)) {
+                static bool module_select[IM_ARRAYSIZE(list_modules)] = { false };
+                for (int m = 0; m < IM_ARRAYSIZE(list_modules); m++) {
+                    if (!search_bar.PassFilter(list_modules[m]))
+                        continue;
+                    modules = std::find(host.cfg.lle_modules.begin(), host.cfg.lle_modules.end(), list_modules[m]);
+                    if (modules != host.cfg.lle_modules.end())
+                        module_select[m] = true;
+                    else if (modules == host.cfg.lle_modules.end())
+                        module_select[m] = false;
+                    if (ImGui::Selectable(list_modules[m], &module_select[m])) {
+                        if (modules == host.cfg.lle_modules.end())
+                            host.cfg.lle_modules.push_back(list_modules[m]);
+                        if (module_select[m] == false) {
+                            if (modules != host.cfg.lle_modules.end())
+                                host.cfg.lle_modules.erase(modules);
+                        }
+                    }
+                }
+                ImGui::ListBoxFooter();
+            }
+            ImGui::PopItemWidth();
+            ImGui::Spacing();
+            ImGui::TextColored(GUI_COLOR_TEXT, "Search Modules");
+            search_bar.Draw("##search_bar", 200);
```

Comment:
Again, i know you are putting this global in order to reuse, put the search bar variable are created very quick. Just put another one inside here